Wrappers/Python/cil/framework/framework.py

```diff
@@ -40,6 +40,205 @@
 
+from cil.framework.BlockGeometry import BlockGeometry
+
+class Partitioner(object):
+    '''Interface for AcquisitionData to be able to partition itself in a number of sub-sets.
+    
+    This class, by multiple inheritance with AcquisitionData, allows the user to partition the data, 
+    by using the method ``partition``. 
+    The partitioning will generate a ``BlockDataContainer`` with appropriate ``AcquisitionData``.
+
+    '''
+    # modes of partitioning
+    SEQUENTIAL = 'sequential'
+    STAGGERED = 'staggered'
+    RANDOM_PERMUTATION = 'random_permutation'
+
+    def _partition_indices(self, num_batches, indices, stagger=False):
+        """Partition a list of indices into num_batches of indices.
+        Parameters
+        ----------
+        num_batches : int
+            The number of batches to partition the indices into.
+        indices : list of int, int
+            The indices to partition. If passed a list, this list will be partitioned in ``num_batches``
+            partitions. If passed an int the indices will be generated automatically using ``range(indices)``.
+        stagger : bool, default False
+            If True, the indices will be staggered across the batches.
+
+        -------
+        list of list of int
+            A list of batches of indices.
+        """
+
+        # Partition the indices into batches.
+        if isinstance(indices, int):
+            indices = list(range(indices))
+
+        num_indices = len(indices)
+        # sanity check
+        if num_indices < num_batches:
+            raise ValueError(
+                'The number of batches must be less than or equal to the number of indices.'
+            )
+
+        if stagger:
+            batches = [indices[i::num_batches] for i in range(num_batches)]
+
+        else:
+            # we split the indices with floor(N/M)+1 indices in N%M groups
+            # and floor(N/M) indices in the remaining M - N%M groups.
+
+            # rename num_indices to N for brevity
+            N = num_indices
+            # rename num_batches to M for brevity
+            M = num_batches
+            batches = [
+                indices[j:j + math.floor(N / M) + 1] for j in range(N % M)
+            ]
+            offset = N % M * (math.floor(N / M) + 1)
+            for i in range(M - N % M):
+                start = offset + i * math.floor(N / M)
+                end = start + math.floor(N / M)
+                batches.append(indices[start:end])
+
+        return batches
+
+    def _convert_masks_to_BlockGeometry(self, masks):
+        '''Convert a list of boolean masks to a list of BlockGeometry.
+        
+        Parameters
+        ----------
+          masks : list of boolean array
+              A list of boolean masks.
+            
+        Returns
+        -------
+            BlockGeometry
+        '''
+        ags = []
+        for mask in masks:
+            ag = self.geometry.copy()
+            ag.set_angles(self.geometry.angles[mask])
```

Comment:
modify the `angles` directly `ag.config.angle.angle_data`, otherwise we need to get the default units and initial angle that are blasted by `set_angles` `ag.config.angle.angle_data`